### ISSUE TITLE
[BUG/Help] 单机多卡训练（4张v100），学习率下降得很快，在1.5个epoch时，学习率已经为0了。

### ISSUE THREAD

**Issue by FanWan** [THUDM/ChatGLM-6B#1289]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

LR=3e-2
PRE_SEQ_LEN=128
SAVE_MODEL_PATH=chatglm_sft_0625
MASTER_PORT=$(shuf -n 1 -i 10000-65535)

export CUDA_VISIBLE_DEVICES=0,1,2

deepspeed --master_port $MASTER_PORT main.py \
    --deepspeed deepspeed.json \
    --do_train \
    --train_file /share/code/ChatGLM-6B/ptuning/dataset_0625/sft-0625_shuffle.json \
    --do_eval \
    --validation_file /share/code/ChatGLM-6B/ptuning/dataset_0625/eval-0625.json \
    --prompt_column prompt \
    --response_column output \
    --overwrite_cache \
    --model_name_or_path /share/llm_models/glm/chatglm-6b \
    --output_dir output/$SAVE_MODEL_PATH \
    --overwrite_output_dir \
    --max_source_length 1536 \
    --max_target_length 512 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 8 \
    --predict_with_generate \
    --max_steps 1000 \
    --logging_steps 20 \
    --save_steps 100 \
    --learning_rate $LR \
    --pre_seq_len $PRE_SEQ_LEN \
    --quantization_bit 4 >> log/wf_train_0625.log 2>&1 &

采用上面指令用deepspeed进行多卡finetune，学习率下降得特别快，trainer_state.json的内容如下：
 {
      "epoch": 0.65,
      "learning_rate": 0.0156,
      "loss": 1.3569,
      "step": 60
    },
    {
      "epoch": 0.87,
      "learning_rate": 0.010799999999999999,
      "loss": 1.0493,
      "step": 80
    },
    {
      "epoch": 1.09,
      "learning_rate": 0.006,
      "loss": 0.8041,
      "step": 100
    },
    {
      "epoch": 1.31,
      "learning_rate": 0.0012,
      "loss": 0.5933,
      "step": 120
    },
    {
      "epoch": 1.53,
      "learning_rate": 0.0,
      "loss": 0.5045,
      "step": 140
    },
    {
      "epoch": 1.74,
      "learning_rate": 0.0,
      "loss": 0.5137,
      "step": 160
    },

### Expected Behavior

单卡训练的时候，学习率衰减看起来比较正常。

### Steps To Reproduce

直接使用ds_train_finetune.sh跑多卡训练。

### Environment

```markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by surmount1** [comment]
gradient_accumulation_steps 多卡时，这个梯度累计参数小点，学习率就不会快速降为0

**Comment by surmount1** [comment]
多卡时，gradient_accumulation_steps 这个梯度累计参数小点，学习率就不会快速降为0

**Comment by FanWan** [comment]
> 多卡时，gradient_accumulation_steps 这个梯度累计参数小点，学习率就不会快速降为0

那设置成什么比较合适呢？gradient_accumulation_steps设置成2，然后per_device_train_batch_size设置成16，这样batch_size还是32 ? 可以吗，请问！

**Comment by FanWan** [comment]
gradient_accumulation_steps 为啥会影响学习率呢，有点不解，

**Comment by dddddjcole** [comment]
你好 方便分享一下你的deepspeed的配置文件吗？我在做书模型并行时发现两张卡仅仅只用了一张卡，在不适用cpu卸载的情况下
